x-pack/plugins/upgrade_assistant/server/lib/reindexing/index_settings.ts

```diff
@@ -19,6 +31,32 @@ export const transformFlatSettings = (flatSettings: FlatSettings) => {
   return { settings, mappings };
 };
 
+/**
+ * Parses an index name
+ * @param indexName
+ */
+export const parseIndexName = (indexName: string): ParsedIndexName => {
+  const matches = indexName.match(/^([\.])?(.*)$/) || [];
+  const isInternal = matches[1] === '.';
+  const baseName = matches[2];
+
+  const currentVersion = `reindexed-v${CURRENT_MAJOR_VERSION}`;
+
+  // in 5.6 the upgrade assistant appended to the index, in 6.7+ we prepend to
+  // avoid conflicts with index patterns/templates/etc
+  const reindexedMatcher = new RegExp(`(-reindexed-v5$|reindexed-v${PREV_MAJOR_VERSION}-)`, 'g');
+
+  const cleanBaseName = baseName.replace(reindexedMatcher, '');
+
+  return {
+    isInternal,
```

Comment:
I don't think this flag is actually used by anything except tests. Seems like an implementation detail we should hide? If you don't think we should keep it for tests, I think we should at least rename it to `_isInternal` to make it more clear.